In [1]:
import numpy as np
import bokeh as bk

# Thermodynamics Week 2 Homework

Mikko Karkkainen 11740378

### Problem 1

Calculate $C_p^{298}$ for $AlF_3$

### solution:

Let's use the Kellogg method to estimate the contributions of Al and F ions to $C_p$. The charge of an F anion is always -1, so the charge of Al must be +3. According to tables 1 and 2 in the handout: 

$$Al : 14.6 \; \frac{J}{g \cdot atom \cdot K}$$

with cation charge of +3:

$$F^-: 21.3 \; \frac{J}{g \cdot atom \cdot K}$$

The total specific heat capacity at calculated by adding the contributions from all the cations and anions of the compound:

$$C_p^{298}= 14.6 \frac{J}{g \cdot K} + 3 \cdot 21.3 \frac{J}{g \cdot K}$$

In [2]:
cp=14.6 + 3*21.3
print(cp)

78.5


**Answer:** 
$$C_p^{298}=78.5 \frac{J}{g \cdot K}$$

-----------------------------------------------------------------------------

### Problem 2

Calculate the change in $H_T - H_{298}$ for Cu

**Solution**:

Let's make the assumption that temperature $T$ is under $T_M$ , so that we calculate enthalpy change in solid phase only.

Standard enthalpy of formation for Cu:

$$ \Delta H_f^{ \theta } (Cu)= 0$$

G. K. White (1984):

<img src="copper">

Let's fit a polynomial of the fourth order to estimate $C_p$ as a function $T$:

In [3]:
Cp=np.array([0.000743,0.00177,0.00337,0.00582,0.0043,0.0145,0.0213,0.0301,0.0414,0.0555,0.0936,0.149,0.225,0.328,0.462,0.963,1.693,2.64,3.74,6.15,8.595,10.86,12.85,14.56,16.01,18.25,19.87,21.05,21.94,22.63,23.17,23.60,23.94,24.22,24.45,24.88,25.25,25.61,25.91,26.21,26.48,26.73,26.99,27.48,28.04,28.66,29.48,30.53,31.12,32.16])

T=np.empty([50])

x=1
y=0

while x < 10:
    T[y]=x
    x+=1
    y+=1
    
while x < 20:
    T[y]=x
    x+=2
    y+=1
    
while x<40:
    T[y]=x
    x+=5
    y+=1
    
while x <100:
    T[y]=x
    x+=10
    y+=1

while x <300:
    T[y]=x
    x+=20
    y+=1
    
while x <700:
    T[y]=x
    x+=50
    y+=1

while x <1200:
    T[y]=x
    x+=100
    y+=1
    
while x <1350:
    T[y]=x
    x+=50
    y+=1

In [4]:
from bokeh.io import push_notebook
import bokeh.plotting as bkp
import bokeh.models as bkm
from bokeh.io import push_notebook
from ipywidgets import interact
bkp.output_notebook()

Loading BokehJS ...

In [5]:
p = bkp.figure(plot_width=800, plot_height=400, title="Copper Cp (G.K. White 1984)", x_axis_label="Temperature (K)", y_axis_label="Cp (J/(mol K))")

r=np.polyfit(T,Cp,deg=6)
pr=np.poly1d(r)

x=np.linspace(1,1300,1300)

p.line(x,pr(x),color='red',line_dash=[5,5], line_width=1)
p.scatter(T,Cp)

bkp.show(p)

 The equation of the 6th degree fitted polynomial of the Cp function:

In [6]:
print(pr)

            6             5             4             3             2
-2.034e-17 x + 2.631e-13 x - 8.242e-10 x + 1.113e-06 x - 0.0007314 x + 0.2307 x - 2.233


Let us calculate the change in enthalpy using the $C_p$ function we acquired:

$$H_T - \Delta H_f^{ \theta }= H_T = \int_{298}^{T} C_p \: dT = H(T) \; \big|_{298}^T $$

which is equal to:

In [7]:
H=np.poly1d.integ(pr)
H=H-H(298)
print(H)

            7             6             5             4             3
-2.906e-18 x + 4.384e-14 x - 1.648e-10 x + 2.784e-07 x - 0.0002438 x
           2
 + 0.1153 x - 2.233 x - 4963


Now let's plot the enthalpy difference

In [8]:
X=x
Y=H(x)
temp=298

In [9]:
bkp.output_notebook()

Loading BokehJS ...

In [10]:
p=bkp.figure(plot_width=800, plot_height=400, title="Copper change of enthalpy",x_axis_label="Temperature(K)", y_axis_label= "Enthalpy (J/(mol))")

r=p.line(X,Y, line_width=1)

X2=np.array([temp,temp])
Y2=np.array([0,H(temp)])

X3=np.array([1,temp])
Y3=np.array([H(temp),H(temp)])

r2=p.line(X2,Y2,line_width=1,color='red')
r3=p.line(X3,Y3,line_width=1,color='red')

p.set(x_range=bkm.Range1d(1, 1300),y_range=bkm.Range1d(-5000,30000))


In [11]:
def update(temp=298):
    
    X2=np.array([temp,temp])
    Y2=np.array([0,H(temp)])
    
    X3=np.array([1,temp])
    Y3=np.array([H(temp),H(temp)])
    
    print("H_T-H298 is equal to {0}  J/mol at {1} K".format(H(temp),temp))
    
    r2.data_source.data['x']=X2
    r2.data_source.data['y']=Y2
    
    r3.data_source.data['x']=X3
    r3.data_source.data['y']=Y3
    
    push_notebook()

In [12]:
bkp.show(p)

In [13]:
interact(update, temp=(1,1300,1))

H_T-H298 is equal to 10360.060590210775  J/mol at 698 K


$H_T - \Delta H_f^{ \theta }$ for 1 mole of copper atoms in constant pressure is given by the function:

$$-2.906 \cdot 10^{-18} x^7 + 4.384 \cdot 10^{-14} x^6 - 1.648 \cdot 10^{-10} x^5 + 2.784 \cdot 10^{-7} x^4 - 0.0002438 x^3 + 0.1153 x^2 - 2.233 x - 4963$$


-----

## Problem 3.

The balanced equation for the reaction between graphite and water is shown below:

$$ H_2O (g) + x O_2 (g) + (1+2 x) C (graphite) = H_2 (g) + (1+2x) CO (g) $$

Calculate the product gas $(H_2$  and  $CO)$ composition at $1000K$. Assume adiabatic conditions(i.2. no heat loss or gain).